src/cargo/core/dependency.rs

```diff
@@ -182,6 +164,26 @@ impl Dependency {
         }
     }
 
+    pub fn serialized(&self, unstable_flags: &CliUnstable) -> SerializedDependency {
+        SerializedDependency {
+            name: self.package_name(),
+            source: self.source_id(),
+            req: self.version_req().to_string(),
+            kind: self.kind(),
+            optional: self.is_optional(),
+            uses_default_features: self.uses_default_features(),
+            features: self.features().to_vec(),
+            target: self.inner.platform.clone(),
+            rename: self.explicit_name_in_toml(),
+            registry: self.registry_id().as_ref().map(|sid| sid.url().to_string()),
+            path: self.source_id().local_path(),
+            artifact: self.inner.artifact.clone(),
+            public: unstable_flags
+                .public_dependency
+                .then_some(self.inner.public),
```

Comment:
When parsing a manifest, we consider `-Z` or `cargo-features` (`Feature::public_dependency()`).  Should we do that here as well?

Reply:
I'm not sure what considering `Feature::public_dependency()` would look like. Do you have a link to where this happens in the code for parsing the manifest?

Reply:
https://github.com/rust-lang/cargo/blob/64c4b6d320b09de9fc5085dafd99c020f7128821/src/cargo/util/toml/mod.rs#L789-L794 is us considering it during manifest parsing.  The information is in one of the related types (manifest? package?).  Unsure if you have access to it all to pass it where its needed.

Reply:
Addressed :+1: